packages/cli/src/util/input/edit-project-settings.ts

```diff
@@ -2,45 +2,88 @@ import inquirer from 'inquirer';
-  { name: 'Development Command', value: 'devCommand' },
+const settingMap = {
+  buildCommand: 'Build Command',
+  devCommand: 'Development Command',
+  commandForIgnoringBuildStep: 'Ignore Command',
+  installCommand: 'Install Command',
+  outputDirectory: 'Output Directory',
+  framework: 'Framework',
+} as const;
+type ConfigKeys = keyof typeof settingMap;
+const settingKeys = Object.keys(settingMap).sort() as unknown as readonly [
+  ConfigKeys
 ];
 
+export type PartialProjectSettings = Pick<ProjectSettings, ConfigKeys>;
+
 export default async function editProjectSettings(
   output: Output,
   projectSettings: PartialProjectSettings | null,
   framework: Framework | null,
-  autoConfirm?: boolean
+  autoConfirm: boolean,
+  localConfigurationOverrides: PartialProjectSettings | null
 ): Promise<ProjectSettings> {
-  // create new settings object, missing values will be filled with `null`
+  // Create initial settings object defaulting everything to `null` and assigning what may exist in `projectSettings`
   const settings: ProjectSettings = Object.assign(
-    { framework: null },
+    {
+      buildCommand: null,
+      devCommand: null,
+      framework: null,
+      commandForIgnoringBuildStep: null,
+      installCommand: null,
+      outputDirectory: null,
+    },
     projectSettings
   );
 
-  for (let field of fields) {
-    settings[field.value] =
-      (projectSettings && projectSettings[field.value]) || null;
+  // Start UX by displaying (and applying) overrides. They will be referenced throughout remainder of CLI.
+  if (localConfigurationOverrides) {
+    // Apply local overrides (from `vercel.json`)
+    for (const setting of settingKeys) {
+      const localConfigValue = localConfigurationOverrides[setting];
+      if (localConfigValue) settings[setting] = localConfigValue;
+    }
+
+    output.print('Local settings detected in vercel.json:\n');
+
+    // Print provided overrides including framework
+    for (const setting of settingKeys) {
+      const override = localConfigurationOverrides[setting];
+      if (override) {
+        output.print(
+          `${chalk.dim(
+            `- ${chalk.bold(`${settingMap[setting]}:`)} ${override}`
+          )}\n`
+        );
+      }
+    }
+
+    // If framework is overridden, set it to the `framework` parameter and let the normal framework-flow occur
+    if (localConfigurationOverrides.framework) {
+      framework = (frameworks.find(
+        _framework => _framework.slug === localConfigurationOverrides.framework
+      ) ?? null) as Framework | null; // if the entered framework is not found and set to `null` we will return early
+
+      if (framework) {
+        output.print(
+          `Merging default project settings for framework ${framework.name}. Previously listed overrides are prioritized.\n`
+        );
+      }
+    }
   }
 
-  // skip editing project settings if no framework is detected
+  // return early if the framework is null.
```

Comment:
This comment is a bit ambiguous because "null" could be the framework object or the slug which have two different meanings

I think the previous comment was better so lets use that

Reply:
Fixed in [662fcc1](https://github.com/vercel/vercel/pull/7811/commits/662fcc1a40ab4b848b54fa208da4a5a93cea8b39) along with a couple other things

Reply:
lgtm! ty!